File: backend/app/schemas/anexo.py
from datetime import datetime
from enum import Enum
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.schemas.common import PaginatedResponse


class AnexoOrigem(str, Enum):
    USUARIO = "USUARIO"
    SISTEMA = "SISTEMA"


class AnexoBase(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id_projeto: int
    file_type: str
    nome_arquivo: str
    tamanho_bytes: int
    origem: AnexoOrigem
    created_by: Optional[int] = None
    created_at: datetime


class AnexoResponse(AnexoBase):
    id: str


class AnexoListResponse(PaginatedResponse[AnexoResponse]):
    pass


class AnexoUploadResponse(AnexoResponse):
    pass

File: frontend/src/types/anexo.ts
export type AnexoOrigem = 'USUARIO' | 'SISTEMA'

export interface Anexo {
  id: string
  id_projeto: number
  file_type: string
  nome_arquivo: string
  tamanho_bytes: number
  created_at: string
  created_by: number | null
  origem: AnexoOrigem
}

export interface AnexoListParams {
  origem?: AnexoOrigem
  page?: number
  per_page?: number
}
